src/main/java/spoon/reflect/visitor/ImportScannerImpl.java

```diff
@@ -72,19 +72,17 @@
 	public <T> void visitCtFieldReference(CtFieldReference<T> reference) {
 		enter(reference);
 		scan(reference.getDeclaringType());
-		// scan(reference.getType());
 		exit(reference);
 	}
 
 	@Override
 	public <T> void visitCtExecutableReference(
 			CtExecutableReference<T> reference) {
 		enter(reference);
-		if (reference.getDeclaringType() != null
-				&& reference.getDeclaringType().getDeclaringType() == null) {
-			addImport(reference.getDeclaringType());
+		// constructor call
+		if (reference.getSimpleName().equals("<init>")) {
+			scan(reference.getDeclaringType());
 		}
-		scan(reference.getType());
```

Comment:
If you add this back, it will make the added test to fail (it will import `internal3.Bar` however it shouldn't.)